diff --git a/cfg.c b/cfg.c
index f820a1a..74a7c88 100644
--- a/cfg.c
+++ b/cfg.c
@@ -662,9 +662,9 @@ static void cfg_dump(void)
 
 int cfg_init(const char *file)
 {
-    if (cfg_check_source(file) == -1) {
-	return -1;
-    }
+//    if (cfg_check_source(file) == -1) {
+//	return -1;
+//    }
 
     if (cfg_read(file) < 0)
 	return -1;
diff --git a/widget_ttf.c b/widget_ttf.c
index 5c0e88c..13784ec 100644
--- a/widget_ttf.c
+++ b/widget_ttf.c
@@ -72,6 +72,7 @@ static void widget_ttf_render(const char *Name, WIDGET_TTF * Image)
 	gdImagePtr gdImage;
 
 	int color;
+	int debugborder;
 	int trans;
 	int brect[8];
 	int _width,_height;
@@ -80,6 +81,9 @@ static void widget_ttf_render(const char *Name, WIDGET_TTF * Image)
 	unsigned char r,g,b,a;
 	char *fcolor;
 	char *text;
+	char *dborder;
+	int max_height;
+	int max_width;
 	double size;
 	char *font,*err,*align;
 
@@ -122,9 +126,12 @@ static void widget_ttf_render(const char *Name, WIDGET_TTF * Image)
 			do
 			{
 				size--;
+				//err = gdImageStringFT(NULL,&brect[0],0,font,size,0.,0,0,"Äp");
+				max_height = brect[3]-brect[7];
 				err = gdImageStringFT(NULL,&brect[0],0,font,size,0.,0,0,text);
+				max_width = brect[2]-brect[6];
 			}
-			while (((brect[2]-brect[6] + 6) > _width) || ((brect[3]-brect[7] + 6) > _height));
+			while ((max_width > _width) || (max_height > _height));
 			x = _width;
 			y = _height;
 		}
@@ -139,8 +146,8 @@ static void widget_ttf_render(const char *Name, WIDGET_TTF * Image)
 			}
 			else
 			{
-				x = brect[2]-brect[6] + 6;
-				y = brect[3]-brect[7] + 6;
+				x = brect[2]-brect[6];
+				y = brect[3]-brect[7];
 			}
 		}
 		Image->gdImage = gdImageCreateTrueColor(x,y);
@@ -180,12 +187,13 @@ static void widget_ttf_render(const char *Name, WIDGET_TTF * Image)
 			switch (toupper(align[0]))
 			{
 			case 'R':
-				x = _width - brect[2] - brect[6] - 3;
+				x = _width - brect[2] - brect[6];
 				break;
 			case 'L':
-				x = 3 - brect[6];
+				x = brect[6];
 				break;
 			case 'C':
+			case 'M':
 				x = (_width - brect[2]-brect[6])/2 - brect[6];
 				break;
 			default:
@@ -193,8 +201,36 @@ static void widget_ttf_render(const char *Name, WIDGET_TTF * Image)
 				break;
 			}
 		else
-			x = 3 - brect[6];
-		y = 3 - brect[7];
+			x = brect[6];
+
+		err = gdImageStringFT(NULL,&brect[0],0,font,size,0.,0,0,"Äp");
+		y = (_height - brect[1] - brect[7])/2;
+
+		dborder = P2S(&Image->debugborder);
+
+		if (strlen(dborder) == 8)
+		{
+			l = strtoul(dborder, &e, 16);
+			r = (l >> 24) & 0xff;
+			g = (l >> 16) & 0xff;
+			b = (l >> 8) & 0xff;
+			a = (l & 0xff) /2;
+
+			debugborder = gdImageColorAllocateAlpha(Image->gdImage, r, g, b, a);
+		}
+		else
+		{
+			l = strtoul(dborder, &e, 16);
+			r = (l >> 16) & 0xff;
+			g = (l >> 8) & 0xff;
+			b = l & 0xff;
+
+			debugborder = gdImageColorAllocate(Image->gdImage, r, g, b);
+		}
+
+		if (debugborder)
+			gdImageRectangle(Image->gdImage, 1, 1, _width-1, _height-1, debugborder);
+
 		err = gdImageStringFT(Image->gdImage,&brect[0],color,font,size,0.0,x,y,text);
 
 	}
@@ -281,6 +316,7 @@ static void widget_ttf_update(void *Self)
 		property_eval(&Image->_width);
 		property_eval(&Image->_height);
 		property_eval(&Image->align);
+		property_eval(&Image->debugborder);
 
 		/* render image into bitmap */
 		widget_ttf_render(W->name, Image);
@@ -336,6 +372,7 @@ int widget_ttf_init(WIDGET * Self)
 		property_load(section, "width", "0", &Image->_width);
 		property_load(section, "height", "0", &Image->_height);
 		property_load(section, "align", "C", &Image->align);
+		property_load(section, "debugborder", "000000", &Image->debugborder);
 
 		/* sanity checks */
 		if (!property_valid(&Image->font))
@@ -392,6 +429,7 @@ int widget_ttf_quit(WIDGET * Self)
 				property_eval(&Image->_width);
 				property_eval(&Image->_height);
 				property_eval(&Image->align);
+				property_free(&Image->debugborder);
 
 				free(Self->data);
 				Self->data = NULL;
diff --git a/widget_ttf.h b/widget_ttf.h
index c088ea9..b2e7556 100644
--- a/widget_ttf.h
+++ b/widget_ttf.h
@@ -46,6 +46,7 @@ typedef struct WIDGET_TTF
 	PROPERTY _width;		/* scale font to witdh */
 	PROPERTY _height;		/* scale font to height */
 	PROPERTY align;		/* align font to L/C/R */
+	PROPERTY debugborder;		/* outer line color */
 } WIDGET_TTF;
 
 extern WIDGET_CLASS Widget_Truetype;
